tests/test_glm_runs.py

```diff
@@ -0,0 +1,61 @@
+import jax
+from neurostatslib.glm import GLM
+from neurostatslib.basis import MSplineBasis
+import numpy as np
+
+class DimensionMismatchError(Exception):
+    """Exception raised for dimension mismatch errors."""
+
+    def __init__(self, message):
+        self.message = message
+        super().__init__(self.message)
+
+def test_setup_Mbasis():
+    """
+    Minimal test for MBasis definition.
+
+    Returns
+    -------
+    None
+
+    Raises
+    ------
+    DimensionMismatchError
+        If the output basis matrix has mismatched dimensions with the specified basis functions or window size.
+
+    Notes
+    -----
+    This function performs a minimal test for defining the MBasis by generating basis functions using different orders.
+    It checks if the output basis matrix has dimensions that match the specified number of basis functions and window size.
+    """
+    n_basis = 6
+    window = 100
+    for order in range(1,6):
+        spike_basis = MSplineBasis(n_basis_funcs=n_basis, window_size=window, order=order)
+        spike_basis_matrix = spike_basis.gen_basis_funcs(np.arange(100))
+        if spike_basis_matrix.shape[0] != n_basis:
+            raise DimensionMismatchError(f"The output basis matrix has {spike_basis_matrix.shape[1]} time points, while the number of basis specified is {n_basis}. They must agree.")
+
+
+        if spike_basis_matrix.shape[1] != window:
+            raise DimensionMismatchError(f"The output basis basis matrix has {spike_basis_matrix.shape[1]} window size, while the window size specified is {window}. They must agree.")
+
+    return
+
+if __name__ == "__main__":
```

Comment:
I'm confused about this ifmain block -- my understanding of `pytest` is that it runs all the `test_` functions within a `test_*py` script, so what's this doing?

Reply:
Specifically, when I run `pytest`, two tests are run, one from each file. And it's not the ifmain block

Reply:
turned everything in a def test_ so that it is run as a test.
I had to comment the plotting otherwise pytest would not terminate until figures were closed

Reply:
Ah, yeah the way to handle this is just to call `plt.close('all')` at the end of the test, [e.g.](https://github.com/LabForComputationalVision/plenoptic/blob/main/tests/test_display.py#L102)

Reply:
locally, it doesn't seem to move to the next line (basically it halts the execution and never gets to plt.close).
I figured out a solution: you need to set 'Agg' as the backend, which doesn't start an event loop like qt does